front/app/containers/Admin/projects/project/nativeSurvey/FormResults/FormResultsQuestion/TextQuestion/Analysis/AnalysisInsights.tsx

```diff
@@ -214,31 +235,56 @@ const Question = ({
   );
 };
 
-const AnalysisInsights = ({ analyses }: { analyses: IAnalysisData[] }) => {
-  const { formatMessage } = useIntl();
+const AnalysisInsights = ({ analysis }: { analysis: IAnalysisData }) => {
+  const [automaticSummaryCreated, setAutomaticSummaryCreated] = useState(false);
   const [selectedInsightIndex, setSelectedInsightIndex] = useState(0);
-  const result = useAnalysisInsightsWithIds({
-    analysisIds: analyses?.map((a) => a.id) || [],
+
+  const { formatMessage } = useIntl();
+
+  const { data: inputs } = useInfiniteAnalysisInputs({
+    analysisId: analysis.id,
+    queryParams: {},
   });
+  const { data: insights } = useAnalysisInsights({
+    analysisId: analysis.id,
+  });
+  const { mutate: addAnalysisSummary } = useAddAnalysisSummary();
+
+  const inputCount = inputs?.pages[0].meta.filtered_count || 0;
+  const selectedInsight = insights?.data[selectedInsightIndex];
 
-  const insights = result
-    .flatMap(({ data }, i) =>
-      data?.data.map((insight) => ({
-        analysisId: analyses[i].id,
-        relationship: insight.relationships.insightable.data,
-      }))
-    )
-    .filter((relationship) => relationship !== undefined) as AnalysisInsight[];
+  // Create a summary if there are no insights yet
 
-  const selectedInsight = insights[selectedInsightIndex];
+  useEffect(() => {
+    if (
+      analysis.id &&
+      insights?.data.length === 0 &&
+      !automaticSummaryCreated &&
+      inputCount > 10
+    ) {
+      setAutomaticSummaryCreated(true);
+      addAnalysisSummary({
+        analysisId: analysis.id,
+        filters: {
+          input_custom_field_no_empty_values: true,
+        },
+      });
+    }
+  }, [
+    analysis.id,
+    addAnalysisSummary,
+    insights,
+    automaticSummaryCreated,
+    inputCount,
+  ]);
 
-  if (insights.length === 0) {
+  if (!insights || insights.data.length === 0) {
```

Comment:
Could we use `insights?.data?.length === 0` instead? Or do we need `!insights` in this check?

Reply:
We do because we need to check for both when insights is undefined and when it is defined but the length is 0 and in both cases we want to return null.

Reply:
Gotchya! Thanks :)